apis/keda/v1alpha1/scaledjob_types.go

```diff
@@ -111,8 +113,19 @@ func init() {
 // MaxReplicaCount returns MaxReplicaCount
 func (s ScaledJob) MaxReplicaCount() int64 {
 	if s.Spec.MaxReplicaCount != nil {
-		return int64(*s.Spec.MaxReplicaCount)
+		return int64(*s.Spec.MaxReplicaCount) - s.MinReplicaCount()
```

Comment:
Without `- s.MinReplicaCount()` Keda would scale to maxReplicaCount + minReplicaCount amount of jobs because the actual scaling logic/ scaling strategy knows nothing about the minReplica scaling. This can be seen greatly when removing `- s.MinReplicaCount()` and running the e2e test.